pgxn/neon/control_plane_connector.c

```diff
@@ -501,9 +501,16 @@ HandleCreateDb(CreatedbStmt *stmt)
 
 	entry->type = Op_Set;
 	if (downer && downer->arg)
-		entry->owner = get_role_oid(defGetString(downer), false);
+        {
+		const char *owner_name = defGetString(downer);
+		if(strcmp(owner_name, "neon_superuser") == 0)
+			elog(ERROR, "Can't create a database with owner neon_superuser");
```

Comment:
NIT: I think that Postgres code style is to start errors with lowercase

Reply:
More important is that you also need to handle ALTER DATABASE, as it can change owner too. Hm, I actually don't see that we handle it in the `NeonProcessUtility()`

Reply:
Good point. I made it so that it blocks it. We handle it in `HandleAlterOwner`